core/view-change.go

```diff
@@ -128,3 +153,71 @@ func validateMessageLog(replicaID uint32, view uint64, log messages.MessageLog,
 
 	return nil
 }
+
+func makeViewChangeProcessor(collect viewChangeCollector, processNewViewCert newViewCertProcessor) viewChangeProcessor {
+	var lock sync.Mutex
+
+	return func(vc messages.ViewChange) (new bool, _ error) {
+		lock.Lock()
+		defer lock.Unlock()
+
+		new, nvCert := collect(vc)
+		if nvCert != nil {
+			processNewViewCert(vc.NewView(), nvCert)
+		}
+
+		return
+	}
+}
+
+func makeViewChangeCollector(id, commitCertSize, n uint32) viewChangeCollector {
+	var (
+		view      uint64
+		collected = make(messages.NewViewCert, 1, n-commitCertSize+1)
```

Comment:
Oh, the capacity should be `n-commitCertSize`

Reply:
The capacity is correct, see [here](https://github.com/hyperledger-labs/minbft/pull/230/files/45aa5e4aa2cf5a145e64a9af6c67b366cab016c1..db5ca76d3b2750be3a942fe9bd1285d21b227898#r789942736).